WordPress/src/main/java/org/wordpress/android/ui/people/PersonDetailFragment.java

```diff
@@ -106,4 +157,104 @@ public void setPersonDetails(long personID, int localTableBlogID) {
+            mRoleContainer.setAlpha(0.5f);
+        }
+    }
+
+    @Override
+    public void onClick(View v) {
+        Context context = getActivity();
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(context, R.style.Calypso_AlertDialog);
+        builder.setTitle(R.string.role);
+        builder.setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                Person person = PeopleTable.getPerson(mPersonID, mLocalTableBlogID);
+                if (person != null) {
+                    // reset the selected role since the dialog is cancelled
+                    mSelectedRole = person.getRole();
+                }
+            }
+        });
+        builder.setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                if (mListener != null) {
+                    mListener.onRoleChanged(mPersonID, mLocalTableBlogID, mSelectedRole);
+                }
+            }
+        });
+
+        final String[] roles = getResources().getStringArray(R.array.roles);
+        ArrayAdapter<String> roleAdapter = new RoleListAdapter(context, R.layout.role_list_row, roles);
+        builder.setAdapter(roleAdapter, null);
+
+        builder.show();
+    }
+
+    private class RoleListAdapter extends ArrayAdapter<String> {
+        public RoleListAdapter(Context context, int resource, String[] objects) {
+            super(context, resource, objects);
+        }
+
+        @Override
+        public View getView(final int position, View convertView, ViewGroup parent) {
+            if (convertView == null) {
+                convertView = View.inflate(getContext(), R.layout.role_list_row, null);
+            }
+
+            final RadioButton radioButton = (RadioButton) convertView.findViewById(R.id.radio);
+            TextView mainText = (TextView) convertView.findViewById(R.id.role_label);
+            String role = getItem(position);
+            mainText.setText(role);
+
+            if (radioButton != null) {
+                radioButton.setChecked(mSelectedRole.equalsIgnoreCase(role));
+                radioButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        changeSelection(position);
+                    }
+                });
+            }
+
+            convertView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    changeSelection(position);
+                }
+            });
+
+            return convertView;
+        }
+
+        private void changeSelection(int position) {
+            mSelectedRole = getItem(position);
```

Comment:
This doesn't seem to work well. The variable is kept between invocations of the Role dialog leading to the previously selected role to be checked, even if the previous dialog was dismissed without "OK".

While at it, does the `mSelectedRole` variable really need to be Fragment global? It seems to belong more into the `RoleListAdapter` and not even be directly exposed as all.